animations/test/misc/hero_animation_test.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:animations/src/misc/hero_animation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+Widget createHeroAnimationDemoScreen() => MaterialApp(
+      home: HeroAnimationDemo(),
+    );
+
+void main() {
+  group('Hero Animation Tests', () {
+    testWidgets('Heigth of Container changes on Tap', (tester) async {
+      await tester.pumpWidget(createHeroAnimationDemoScreen());
+
+      // Get the initial Container.
+      final initialContainer =
+          tester.firstWidget(find.byType(Container)) as Container;
+
+      // Get height of Container.
+      // In this case minHeight is equal to maxHeight.
+
+      var initialHeight =
+          initialContainer.constraints.heightConstraints().maxHeight;
+
+      // Tap on the GestureDetector.
+      await tester.tap(find.byType(GestureDetector));
+      await tester.pumpAndSettle();
+
+      // Get the final Container.
+      final finalContainer =
+          tester.firstWidget(find.byType(Container)) as Container;
+
+      // Get height of Container.
+      // In this case minHeight is equal to maxHeight.
+
+      var finalHeight =
+          finalContainer.constraints.heightConstraints().maxHeight;
+
+      expect(
+        initialHeight,
+        lessThan(finalHeight),
+      );
+    });
+
+    testWidgets('Width of Container changes on Tap', (tester) async {
```

Comment:
I think comparing both height & width in separate test cases is not required. It would be more efficient to check for complete size instead.

Reply:
Since, the `Container` is constrained on both sides. I think a better approach will be to take the area and test it against final area of `Container`. What are your view about this?